fs/vfs/fs_lock.c

```diff
@@ -774,6 +808,8 @@ void file_closelk(FAR struct file *filep)
     }
 
   nxmutex_unlock(&g_protect_lock);
+out:
+  lib_put_pathbuffer(path);
```

Comment:
Maybe better
```suggestion
out:
  nxmutex_unlock(&g_protect_lock);
out_free:
  lib_put_pathbuffer(path);
```
and remove `nxmutex_unlock(&g_protect_lock);` before `goto out;`?

Reply:
Done